### ISSUE TITLE
Move newpb out of trunk into a branch or separate package

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1999]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1999|
|Type|task|
|Created|2006-08-15 17:00:04Z|

Based on the current failing state of the tests and the insufficient maintenance resources, new pb should be developed outside of Twisted trunk.

Putting this back into a branch is perfectly acceptable, but it may be useful to split it out into a separate (not sub-) project where releases can be made more easily and frequently to help get it some momentum.

<details><summary>Searchable metadata</summary>

```
trac-id__1999 1999
type__task task
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__pb pb
keywords__ 
time__1155661204000000 1155661204000000
changetime__1159730166000000 1159730166000000
version__None None
owner__ 
cc__radix cc__warner cc__TimothyFitz
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)<a name="note_1"></a>|@warner set status to `assigned`|
|-|-|

I agree. I intend to move it to a separate sub-project, with a new package name (not underneath twisted.*), at something like svn://svn.twistedmatrix.com/svn/NAME/trunk . That would help development and installation too, since it wouldn't have to be installed under twisted/ to be usable.

I'm trying to come up with a reasonable interim name now.

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)<a name="note_2"></a>|@warner set owner to @exarkun<br>@warner set status to `new`|
|-|-|

I've moved newpb out to Twisted/sandbox/warner/Foolscap .
I've created Twisted/branches/newpb-rename-1999 to handle the newpb-ectomy. newpb is now gone
from that branch, and it's ready for review to make sure I didn't miss any traces of it.

I'll be releasing Foolscap-0.0.2 shortly, once I've gone through and finished renaming the internals and updating the docs.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix set owner to @warner|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [18300]) Merge newpb-rename-1999

Author: warner
Reviewer: radix
Fixes [#1999](https://github.com/twisted/twisted/issues/1999)

Take newpb out of Twisted proper.  Future development will take place somewhere
else in the repository.

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)<a name="note_5"></a>|@warner set status to `reopened`|
|-|-|

thanks for doing the merge.. it looks like the (empty) twisted/pb directory is still left over. Shall I create a new branch to remove this directory and begin the review process again?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Reverted r18300, deleted twisted/pb/ from newpb-rename-1999.  Can some people take a look over the branch and see if there's anything else left that needs to be removed?

**Comment by twisted-trac** [comment]
|<img alt="TimothyFitz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|TimothyFitz set owner to TimothyFitz|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="TimothyFitz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|TimothyFitz set owner to @warner|
|-|-|

All tests pass on major platforms, and I can't find any reference to the newpb code. Things look good (passes review).

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)<a name="note_10"></a>|@warner set status to `closed`|
|-|-|

(In [18362]) Remove newpb (twisted/pb) from trunk.

Author: warner

Reviewer: TimothyFitz

Fixes [#1999](https://github.com/twisted/twisted/issues/1999)

It has been moved into warner's sandbox, and renamed Foolscap for now.
See http://twistedmatrix.com/trac/wiki/FoolsCap for details.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
